surfer/viz.py

```diff
@@ -992,12 +993,22 @@ def add_data(self, array, min=None, max=None, thresh=None,
             min value in colormap (uses real min if None)
         max : float
             max value in colormap (uses real max if None)
+        mid : float
+            intermediate value in colormap (middle between min and max if None)
         thresh : None or float
             if not None, values below thresh will not be visible
+        center : float or None
+            if not None, center of a divergent colormap, changes the meaning of
+            min, max and mid, see ``scale_data_colormap`` for further info
+        transparent : bool
+            whether colors from mid to min should become increasingly 
+            transparent
         colormap : string, list of colors, or array
             name of matplotlib colormap to use, a list of matplotlib colors,
             or a custom look up table (an n x 4 array coded with RBGA values
-            between 0 and 255).
+            between 0 and 255), the default "auto" chooses a default divergent 
+            colormap, if "center" is given, otherwise a default sequential 
```

Comment:
might as well list what they will be, e.g. '... default divergent colormap ("icefire") if "center" is given, otherwise ...'

Reply:
One reason not to do that is that encoding defaults in text that doesn't automatically update with the code is asking for trouble later if things change.

Reply:
That's true. There's a tradeoff to be made there. In this case, once we choose defaults it's backward-incompatible to change them, so the change is likely to be invasive enough we'd notice that we need to update the defaults. I think the informativeness to the user is worth it in this case -- what do you think?

Reply:
In this case, breaking backwards compatibility wouldn't cause any existing code to stop working or produce incorrect results, it would simply look different. So in the future, if we decide there is a better default colormap to use, I would feel no compunction about the change.

Reply:
(That said, the worst thing that would happen if the docs don't get updated is some minor confusion for an already curious user, so maybe it's fine).

Reply:
>  wouldn't cause any existing code to stop working or produce incorrect results, it would simply look different. 

For scripts that use PySurfer to make figures for publication (hopefully there are many of these!) this would be problematic, so I wouldn't want to do this without a deprecation cycle.

Reply:
Well, you only publish once, and it will be extremely obvious if/when it changes, so that doesn't strike me as a problem. With this sort of parameter my opinion is that not specifying it is "asking for the default", which may be subject to change. In other words, best practices for publication scrips that you really don't want to change is to encode the values you really want even when they correspond to the default at the time you write them.

Overly stubborn resistance to changing default colormap even when you know it's bad is what caused the sourge of jet after all :)

Anyway it's an orthogonal issue and I'm fine saying in the docs what the default is. Maybe add a comment in the code that the docstring should be updated if the defaults are changed.